### PR TITLE
config: add self observability metrics to the collector config

### PR DESCRIPTION
**Pull request by ridwanmsharif**
This change adds the following self-observability metrics for the collector:
 - otelcol_process_uptime: Uptime of the collector
 - otelcol_process_memory_rss: Memoru usage of the collector
 - otelcol_grpc_io_client_completed_rpcs: Request count to the Google Observability API
 - otelcol_googlecloudmonitoring_point_count: Metric points writted to Cloud Monitoring

These metrics were chosen for now as they are collected for the Ops Agent and the Cloud Run GMP sidecar as well.

We can expand this list in future versions of the manifest.

### CODE REVIEW

File: k8s/overlays/test/fixture.json [hunk 1]
```diff
@@ -1 +1 @@
...tSpanId":"a514c25ace3f0a5a","name":"/hello","kind":"SPAN_KIND_SERVER","startTimeUnixNano":"1658413463217165380","endTimeUnixNano":"1658413463225568609","attributes":[{"key":"net.transport","value":{"stringValue":"ip_tcp"}},{"key":"net.peer.ip","value":{"stringValue":"127.0.0.1"}},{"key":"net.peer.port","value":{"intValue":"32878"}},{"key":"net.host.ip","value":{"stringValue":"0.0.0.0"}},{"key":"net.host.port","value":{"intValue":"7080"}},{"key":"http.method","value":{"stringValue":"GET"}},{"key":"http.target","value":{"stringValue":"/hello"}},{"key":"http.server_name","value":{"stringValue":"/hello"}},{"key":"http.user_agent","value":{"stringValue":"Go-http-client/1.1"}},{"key":"http.scheme","value":{"stringValue":"http"}},{"key":"http.host","value":{"stringValue":"0.0.0.0:7080"}},{"key":"http.flavor","value":{"stringValue":"1.1"}},{"key":"server-attribute","value":{"stringValue":"foo"}},{"key":"baggage key:method","value":{"stringValue":"repl"}},{"key":"baggage key:client","value":{"stringValue":"cli"}},{"key":"http.wrote_bytes","value":{"intValue":"11"}},{"key":"http.status_code","value":{"intValue":"200"}}],"status":{}}]}],"schemaUrl":"https://opentelemetry.io/schemas/1.10.0"}]}
```
Comment (damemi) [inline k8s/overlays/test/fixture.json:1]: any idea why the change here? Looks like `host.name` got dropped, not sure if there's any other changes

Comment (ridwanmsharif) [reply k8s/overlays/test/fixture.json:1]: when I run `make generate` on `main` I get the same result. Maybe when we checked in the change, we forgot to run `make generate` one last time?

**Review by damemi — APPROVED** [review]
